pkg/reconciler/common/ha.go

```diff
@@ -30,6 +30,18 @@ func haUnSupported(obj base.KComponent) sets.String {
 	)
 }
 
+// When Deployment has HPA, the replicas should be controlled by HPA's minReplicas instead of operator.
+// Hence, skip changing the spec.replicas in deployment directory for these Deployments.
+func hasHorizontalPodAutoscaler(obj base.KComponent) sets.String {
+	return sets.NewString(
+		"webhook",
+		"activator",
+		"eventing-webhook",
+		"mt-broker-ingress",
+		"mt-broker-filter",
+	)
+}
```

Comment:
I think downstream we remove some HPA config, so I think we should not hardcode the list

Reply:
Do you have any other good/simple way to solve it without the hardcode?
While I also think that the hardcode the list is not good, I wonder that upstream does not need to consider about downstream's their own customization".

Reply:
Anyway I am not familiar with eventing so I dropped the eventing components from the list.